pkg/service/auth.go

```diff
@@ -47,8 +48,10 @@ func (self *AuthService) Check(ctx context.Context, req *envoy_auth.CheckRequest
 	}
 	authServiceLog.Info("Check()", "reqJSON", string(reqJSON))
 
-	// service config
-	host := req.Attributes.Request.Http.Host
+	// If the host header contains a port, remove it, we only want to match the host part.
+	splitHost := strings.Split(req.Attributes.Request.Http.Host, ":")
```

Comment:
What if we rolled back the change in the Service controller (keeping the key as stated by the developer), and here we first attempted a lookup with the exact host info passed by the proxy (including port number if present); if it doesn't find one, we then tried again without the port number?